rocAL/rocAL_hip/CMakeLists.txt

```diff
@@ -19,6 +19,14 @@
 # THE SOFTWARE.
 
 cmake_minimum_required(VERSION 3.10)
+
+if(NOT DEFINED CMAKE_CXX_COMPILER AND EXISTS "${ROCM_PATH}/bin/amdclang++")
+    set(CMAKE_C_COMPILER ${ROCM_PATH}/bin/amdclang)
+    set(CMAKE_CXX_COMPILER ${ROCM_PATH}/bin/amdclang++)
+endif()
```

Comment:
what about the else case
elseif(NOT DEFINED CMAKE_CXX_COMPILER AND NOT EXISTS "${ROCM_PATH}/bin/amdclang++")